libcxx/test/libcxx/system_reserved_names.gen.py

```diff
@@ -162,4 +163,25 @@
 #define refresh SYSTEM_RESERVED_NAME
 
 #include <{header}>
+
+// Note: We can't include test_macros.h in this test so we use the libc++ version macro.
+#ifndef _LIBCPP_CXX03_LANG
+
+#  define STRINGIFY_IMPL(x) #x
+#  define STRINGIFY(x) STRINGIFY_IMPL(x)
+
+// This is written to work in C++11 constexpr rules
+static constexpr bool is_equal(char const* s1, char const* s2) {{
+  return s1[0] == 0 && s2[0] == 0 ? true :                   // Either both are at the end, or
+              s1[0] != 0 && s2[0] != 0 &&                     // neither is at the end, and
+                  s1[0] == s2[0] && is_equal(s1 + 1, s2 + 1); // their contents are equal
+}}
```

Comment:
Can't we just do `return __builtin_strcmp(s1, s2)`, since this is libc++-internal anyways?

Reply:
Great point, I can actually support C++03 that way. Done.